extra/onnx_ops.py

```diff
@@ -251,4 +251,12 @@ def NegativeLogLikelihoodLoss(input, target, weight=None, ignore_index=None, red
   loss = (-mask * input).sum(axis=1) * (1 if weight is None else weight)  
   if reduction == "mean": return loss.mean() if weight is None else loss.sum() / weight.sum()
   elif reduction == "sum": return loss.sum()
-  return loss.reshape(t_shape) if len(i_shape) != 3 else loss
\ No newline at end of file
+  return loss.reshape(t_shape) if len(i_shape) != 3 else loss
+
+def OneHot(indices, depth, values, axis=-1):
+  indices, rank = (indices.cast(dtypes.float32) < 0).where(indices+depth, indices), len(indices.shape)
+  C = int(depth.lazydata.realize().realized._buf)
```

Comment:
`safe_numpy(depth).item()` will extract the single scalar value

Reply:
Sweet, thank u.